src/views/MarketFilter.tsx

```diff
@@ -29,12 +29,14 @@ export const MarketFilter = ({
     <>
       {!isSearch && (
         <ToggleGroup
-          items={Object.entries(filters).map(([key, value]) => ({
+          items={Object.values(filters).map((value) => ({
             label: stringGetter({ key: MARKET_FILTER_LABELS[value] }),
             value,
           }))}
           value={selectedFilter}
-          onValueChange={onChangeFilter}
+          onValueChange={(newFilter: string) => {
+            onChangeFilter((newFilter || selectedFilter) as MarketFilters);
```

Comment:
is `newFilter` ever expected to be empty or nullable?

Reply:
that's how ToggleGroup is set up currently, when you click on it again, it "removes" the filter